src/core_plugins/kibana/public/dashboard/dashboard_app.js

```diff
@@ -329,8 +327,41 @@ app.directive('dashboardApp', function ($injector) {
         dashboardStateManager.setFullScreenMode(true);
       navActions[TopNavIds.EXIT_EDIT_MODE] = () => onChangeViewMode(DashboardViewMode.VIEW);
       navActions[TopNavIds.ENTER_EDIT_MODE] = () => onChangeViewMode(DashboardViewMode.EDIT);
+      navActions[TopNavIds.SAVE] = () => {
+        const currentTitle = dashboardStateManager.getTitle();
+        const currentDescription = dashboardStateManager.getDescription();
+        const currentTimeRestore = dashboardStateManager.getTimeRestore();
+        const onSave = ({ newTitle, newDescription, newCopyOnSave, newTimeRestore, isTitleDuplicateConfirmed, onTitleDuplicate }) => {
+          dashboardStateManager.setTitle(newTitle);
+          dashboardStateManager.setDescription(newDescription);
+          dashboardStateManager.savedDashboard.copyOnSave = newCopyOnSave;
+          dashboardStateManager.setTimeRestore(newTimeRestore);
+          const saveOptions = {
+            confirmOverwrite: false,
+            isTitleDuplicateConfirmed,
+            onTitleDuplicate,
+          };
+          return $scope.save(saveOptions).then(id => {
+            // If the save wasn't successful, put the original values back.
```

Comment:
Might want to notify the user, too?

Reply:
Rejected save promises are already displayed to users - but I think this should be displayed in a toast notification like other failures. I will create an issue to address.

<img width="2046" alt="screen shot 2018-05-31 at 8 17 14 am" src="https://user-images.githubusercontent.com/373691/40787539-1d332108-64ab-11e8-8b5b-9fc039b0e5d5.png">